src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -1518,10 +1518,14 @@ private object ConvertValue(object value, _SqlMetaData metadata, bool isNull, re
                         value = SqlParameter.CoerceValue(value, mt, out coercedToDataFeed, out typeChanged, false);
                         if (!coercedToDataFeed)
                         {   // We do not need to test for TextDataFeed as it is only assigned to (N)VARCHAR(MAX)
-                            int len = ((isSqlType) && (!typeChanged)) ? ((SqlString)value).Value.Length : ((string)value).Length;
-                            if (len > length / 2)
+                            string str = ((isSqlType) && (!typeChanged)) ? ((SqlString)value).Value : ((string)value);
+                            if (str.Length > length / 2)
                             {
-                                throw SQL.BulkLoadStringTooLong();
+                                if (str.Length > 100)
+                                {
+                                    str = str.Remove(100);
+                                }
+                                throw SQL.BulkLoadStringTooLong(_destinationTableName, metadata.column, str);
```

Comment:
if the above condition is not met, and `str` is not truncated, The `Truncated Value:` would show entire string, is that correct behavior since you're matching it with SQL Server?

Reply:
Yes, SQL Server does not make a difference.

Reply:
Can you give me a query file that you used to test it out - just for confirming the behavior?

Reply:
Check out the announcement blog posts, it demonstrates this behavior https://blogs.msdn.microsoft.com/sql_server_team/string-or-binary-data-would-be-truncated-replacing-the-infamous-error-8152/

Reply:
Oh i see, you might be right, i've read it wrong it seems

Reply:
Thanks for linking the blog post, I was wondering where to begin validation.
So that brings us 2 interesting points for this change:
- This error message is available from SQL Server 2019 (back ported to SQL 2017 CU12 and SQL 2016 SP2 CU but not enabled by default). For SqlClient users this is the only error message applicable to all servers, but I don't see any issue and a good opportunity to address this issue now!
- Magic number **_100_** seems like the max characters _(upon discovery, no official information regarding the same)_ that the error message would contain as "**truncated value**" that prints characters to be inserted in destination column.

My only worry was _[also comes from SQL Server's change, can't really do anything about it]_:
- Printing first **_100_** characters only doesn't solve the purpose of large strings getting truncated, users wouldn't know what's going on for strings longer than 100 characters.

**TODO**: Please add a comment in code for this number truncation to ensure we can address any complaints in future.